packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -600,15 +603,44 @@ class CachedArtifacts implements Artifacts {
   String _getDesktopArtifactPath(Artifact artifact, TargetPlatform platform, BuildMode? mode) {
     // When platform is null, a generic host platform artifact is being requested
     // and not the gen_snapshot for darwin as a target platform.
-    if (artifact == Artifact.genSnapshot) {
-      final String engineDir = _getEngineArtifactsPath(platform, mode)!;
-      return _fileSystem.path.join(engineDir, _artifactToFileName(artifact, _platform));
-    }
-    if (artifact == Artifact.flutterMacOSFramework) {
-      final String engineDir = _getEngineArtifactsPath(platform, mode)!;
-      return _getMacOSEngineArtifactPath(engineDir, _fileSystem, _platform);
+    final String engineDir = _getEngineArtifactsPath(platform, mode)!;
+    switch (artifact) {
+      case Artifact.genSnapshot:
+        return _fileSystem.path.join(engineDir, _artifactToFileName(artifact, _platform));
+      case Artifact.engineDartSdkPath:
+      case Artifact.engineDartBinary:
+      case Artifact.engineDartAotRuntime:
+      case Artifact.dart2jsSnapshot:
+      case Artifact.dart2wasmSnapshot:
+      case Artifact.frontendServerSnapshotForEngineDartSdk:
+      case Artifact.constFinder:
+      case Artifact.flutterFramework:
+      case Artifact.flutterFrameworkDsym:
```

Comment:
Should these be falling through to `_getHostArtifactPath`, not `_getMacOSFrameworkPath`?

Reply:
They should be under `_getMacOSFrameworkPath` since eventually these won't be part of the host build. They really shouldn't have been from the get-go.